### DISCUSSION TITLE
MinIO is distributed under GNU AGPLv3

### DISCUSSION THREAD

**Discussion by filippo82** [nebari-dev/nebari#987] [Q&A]
Hi all,

I see that PR #967 is going to bring MinIO into the game. MinIO is distributed under GNU AGPLv3. Isn't this going to be a problem?

**Comment by viniciusdc** [comment]
c.c @dharhas @costrouc

**Comment by dharhas** [comment]
Yeah this seems to be a major issue. @filippo82 thanks for bringing it up.

**Reply by filippo82** [reply]
Hi @dharhas 👋🏻 I've been trying to find a reference which explicitly mentions using an AGPL-licensed program as a containerised service but I could not find anything useful yet.

However, this item in the FAQ [here](https://www.gnu.org/licenses/gpl-faq.html#IfLibraryIsGPL) states that:

> If a library is released under the GPL (not the LGPL), does that mean that any software which uses it has to be under the GPL or a GPL-compatible license? (#IfLibraryIsGPL)
Yes, because the program actually links to the library. As such, the terms of the GPL apply to the entire combination. The software modules that link with the library may be under various GPL compatible licenses, but the work as a whole must be licensed under the GPL. See also: What does it mean to say a license is “compatible with the GPL”?

Obviously this is not the same thing... but very close I guess.

**Reply by filippo82** [reply]
Ok... just found some interesting content. It seems that using MinIO in the current way (running it from a container) should (IMHO) not be a problem for QHub.

* https://opensource.stackexchange.com/questions/8375/gnu-agpl-gpl-3-0-and-docker-containers
* https://opensource.stackexchange.com/questions/12168/does-using-an-agpl-3-0-software-via-its-rest-api-force-to-disclose-source-from-ca/12186 <- This one even references MinIO 👍🏻

**Reply by dharhas** [reply]
@filippo82 as a short term fix, we are pinning minio to 6.7.4, the last release that was Apache-2.0.

**Comment by Adam-D-Lewis** [comment]
I'm closing old discussions in a repo cleanup effort.  Feel free to re-open if needed.
